### ISSUE TITLE
Build failure while opam installing git.2.1.2 

### ISSUE THREAD

**Issue by kayceesrk** [mirage/ocaml-git#388]
The failure is due to the recent angstrom version 0.14.1 being picked up. See the error below. I confirmed that I can install `git.2.1.2` with `angstrom.0.13.0`.

```
#=== ERROR while compiling git.2.1.2 ==========================================#
# context     2.0.0 | linux/x86_64 | ocaml-base-compiler.4.09.0 | git+https://github.com/ocaml/opam-repository.git
# path        ~/.opam/ocaml-base-compiler.4.09.0/.opam-switch/build/git.2.1.2
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p git -j 4
# exit-code   1
# env-file    ~/.opam/log/git-18768-457407.env
# output-file ~/.opam/log/git-18768-457407.out
### output ###
# 99 |             | Ok v -> v
# [...]
# Error: This pattern matches values of type ('a, 'b) result
#        but a pattern was expected which matches values of type
#          consume:Angstrom.Consume.t -> (Git__.User.A.e, string) result
#     ocamlopt src/git/.git.objs/native/git__Helper.{cmx,o} (exit 2)
# (cd _build/default && /home/kc/.opam/ocaml-base-compiler.4.09.0/bin/ocamlopt.opt -w -40 -g -I src/git/.git.objs/byte -I src/git/.git.objs/native -I /home/kc/.opam/ocaml-base-compiler.4.09.0/lib/angstrom -I /home/kc/.opam/ocaml-base-compiler.4.09.0/lib/astring -I /home/kc/.opam/ocaml-base-compiler.4.09.0/lib/bigarray-compat -I /home/kc/.opam/ocaml-base-compiler.4.09.0/lib/bigstringaf -I /home/[...]
# File "src/git/helper.ml", line 192, characters 6-10:
# 192 |     | Ok _ as v -> v | Error err -> Error.(v @@ Decoder.err_result input err)
#             ^^^^
# Error: This pattern matches values of type ('a, 'b) result
#        but a pattern was expected which matches values of type
#          consume:Angstrom.Consume.t -> (A.e, string) result
```

I confirmed that

**Comment by dinosaure** [comment]
Ah yes, the update is currently on the pipe when `encore` (https://github.com/mirage/encore/pull/22) was updated (but not yet released) with `angstrom.0.14.0`. But we probably should add a constraint in OPAM at least (I believed that it was done).

Thank for your report!

**Comment by kayceesrk** [comment]
Thanks for following up @dinosaure!

**Comment by dinosaure** [comment]
Fixed by #384
